### PR TITLE
EWPP-58: Create "Call for tenders" corporate content type

### PR DESCRIPTION
**Pull request by julien-**
### Change log

- Added:
  - "Call for tenders" corporate content type
  - Possibility to test (Behat) Date list fields
- Updated:
  - Reorder alphabetically [behat.yml.dist](https://github.com/openeuropa/oe_content/pull/237/files#diff-7bde54db60a6e933518d8b61b929edce)
  - Clean copy-paste code in Behat functions for Date fields

### CODE REVIEW

**Comment by julien-** [comment]
Does any admin can remove this useless branches ? :

- EPIC-Project-typed_link
- EPIC-Project-update
- EPIC-Project-update-3371
- EPIC-Project-typed_link
- EPIC-Tenders (I will recreate it without some code ... just pointing into master)

Thank you

**Comment by julien-** [comment]
The **Subject** field is required by Behat tests _content-types.feature_ so I added the field in the form but it is not supposed to be displayed in the mockup (see [Specifications](https://webgate.ec.europa.eu/fpfis/wikis/pages/viewpage.action?pageId=437857073)).

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (drishu) [inline modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml:62]: Please dont add groups here at oe_content level only for fields that belong together in logic, see Projects and Events content type form view

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml
Comment (drishu) [inline modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml:13]: I'm not sure if the label of this should not be "Departments", and then oe_author is missing and has this label, but I posted this questions in the ticket so ignore this comment for now, just here for history

File: tests/features/tender.feature
Comment (drishu) [inline tests/features/tender.feature:5]: and see call for tenders items

File: modules/oe_content_tender/config/install/core.entity_view_display.node.oe_tender.teaser.yml
Comment (drishu) [inline modules/oe_content_tender/config/install/core.entity_view_display.node.oe_tender.teaser.yml:1]: I dont see anything in the ticket about a teaser view

File: modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml
Comment (drishu) [inline modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml:5]: you are missing something like
```suggestion
    - node.type.oe_tender
module:
    - content_translation
third_party_settings:
  content_translation:
    enabled: false
```

File: config/install/core.entity_view_mode.node.card.yml
Comment (julien-) [reply config/install/core.entity_view_mode.node.card.yml]: Fixed by removing file

File: tests/features/tender.feature
Comment (julien-) [reply tests/features/tender.feature]: Fixed

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml]: Fixed

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml]: Fixed

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml]: Fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_publication_date.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_publication_date.yml]: Fixed

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml]: Fixed. Groups in the form display are removed

File: modules/oe_content_tender/config/install/core.entity_view_display.node.oe_tender.teaser.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/core.entity_view_display.node.oe_tender.teaser.yml]: Fixed by removing Teaser view mode

File: modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml
Comment (julien-) [reply modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml:5]: Doing `drush cde oe_content_tender -y` it is not added so I don't think if it is necessary.
Am I wrong ?

Comment (drishu) [reply modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml:5]: I think it is needed, I think that this is a manual change in preparation of ewcms integration, that is probably why it is not added to config devel export

Comment (julien-) [reply modules/oe_content_tender/config/optional/language.content_settings.node.oe_tender.yml:5]: Fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_reference.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_reference.yml:12]: Fixed

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (nagyad) [inline modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml:27]: Let's order the common fields here like we did in other content types.

File: tests/features/tender.feature
Comment (nagyad) [inline tests/features/tender.feature:27]: And

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml
Comment (nagyad) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml:13]: This requires clarification as "Responsible department" name is colliding with the oe_author field.

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_documents.yml
Comment (nagyad) [inline modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_documents.yml:20]: Field instances should always be not translatable by default, while storage fields should.

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_summary.yml
Comment (nagyad) [inline modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_summary.yml:16]: Field instances should always be not translatable by default, while storage fields should.

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_teaser.yml
Comment (nagyad) [inline modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_teaser.yml:16]: Field instances should always be not translatable by default, while storage fields should.

File: tests/Behat/DateFieldContext.php
Comment (drishu) [reply tests/Behat/DateFieldContext.php:42]: it was moved to https://github.com/openeuropa/oe_content/blob/EWPP-58/tests/Behat/DateFieldContext.php#L171

File: tests/Behat/DateFieldContext.php
Comment (julien-) [reply tests/Behat/DateFieldContext.php:161]: I had to create a function for date time select list widget. I have the choice to copy-paste existing code and just change two little things or refactor code.

This function will be a copy-paste of these functions, that are also a copy-paste between them. Same job for only two changes : the label and the CSS selector.
Insofar as this functions are only used in this file, nowhere else, I removed them and I implemented one single function `findDateFields` with two parameters.

The same job is done, just by calling this function with the right label and the right CSS selector.

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_publication_date.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_publication_date.yml]: done

File: modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/core.entity_form_display.node.oe_tender.default.yml:46]: done

File: tests/features/tender.feature
Comment (drishu) [reply tests/features/tender.feature]: fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_documents.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_documents.yml]: fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_summary.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_summary.yml]: fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_teaser.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_teaser.yml]: fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_departments.yml:13]: this was clarified, this label seems to be ok here (is probably wrong in events) and oe_author will not be added to tender content type

**Review by ademarco — CHANGES_REQUESTED** [review]

File: modules/oe_content_tender/config/install/field.storage.node.oe_tender_opening_date.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/field.storage.node.oe_tender_opening_date.yml]: Fixed

File: modules/oe_content_tender/config/install/field.storage.node.oe_tender_deadline.yml
Comment (julien-) [reply modules/oe_content_tender/config/install/field.storage.node.oe_tender_deadline.yml]: Fixed

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (julien-) [reply modules/oe_content_tender/oe_content_tender.info.yml]: Fixed

File: tests/Behat/DateFieldContext.php
Comment (julien-) [reply tests/Behat/DateFieldContext.php]: Fixed

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_tender_opening_date.yml
Comment (imanoleguskiza) [inline modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_tender_opening_date.yml:13]: This should probably be Opening Date?

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

File: modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_tender_opening_date.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/field.field.node.oe_tender.oe_tender_opening_date.yml:13]: this is the label from the ticket, but seems wrong indeed, as specifications indicate "Opening date" (https://webgate.ec.europa.eu/fpfis/wikis/pages/viewpage.action?pageId=437857073)

File: modules/oe_content_tender/config/install/core.base_field_override.node.oe_tender.title.yml
Comment (drishu) [reply modules/oe_content_tender/config/install/core.base_field_override.node.oe_tender.title.yml:13]: are you sure its EWPP-200 ? I don't see any code in there related to this

**Review by imanoleguskiza — APPROVED** [review]

File: modules/oe_content_departments_field/config/install/field.storage.node.oe_departments.yml
Comment (ademarco) [inline modules/oe_content_departments_field/config/install/field.storage.node.oe_departments.yml:16]: Let's use `true` here as we did for the other `skos_concept_entity_reference` fields.

File: modules/oe_content_publication/oe_content_publication.info.yml
Comment (ademarco) [inline modules/oe_content_publication/oe_content_publication.info.yml:8]: Is this a dependency of this module? If it's not we need to remove it.

File: modules/oe_content_tender/README.md
Comment (ademarco) [inline modules/oe_content_tender/README.md:12]: Is this really a required module?

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (ademarco) [inline modules/oe_content_tender/oe_content_tender.info.yml:8]: Do we really need this module here?

File: modules/oe_content_tender/README.md
Comment (ademarco) [inline modules/oe_content_tender/README.md:15]: Is this really a required module?

**Review by ademarco — CHANGES_REQUESTED** [review]

File: modules/oe_content_publication/oe_content_publication.info.yml
Comment (drishu) [reply modules/oe_content_publication/oe_content_publication.info.yml]: couldnt find any ref to it so probably not

**Review by ademarco — APPROVED** [review]

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (yenyasinn) [inline modules/oe_content_tender/oe_content_tender.info.yml:7]: List of dependencies has to be:
```
  - composite_reference:composite_reference
  - drupal:datetime
  - drupal:link
  - oe_content:oe_content
  - oe_content:oe_content_departments_field
  - oe_content:oe_content_reference_code_field
  - oe_content:oe_content_publication
  - oe_media:oe_media
```

Comment (yenyasinn) [reply modules/oe_content_tender/oe_content_tender.info.yml:7]: Or instead of  - oe_content:oe_content_publication has to be oe_content:oe_content_documents_field.

Comment (yenyasinn) [reply modules/oe_content_tender/oe_content_tender.info.yml:7]: Or instead of - oe_content:oe_content_publication has to be oe_content:oe_content_documents_field from EPIC-Project.

Comment (ademarco) [reply modules/oe_content_tender/oe_content_tender.info.yml:7]: You are right @yenyasinn, let's remove this field all at once and create a follow up,  @nagyad will do that. We will add it when project will be merged, as discussed.

**Review by ademarco — CHANGES_REQUESTED** [review]
Blocking as we need to remove the publication document for now, as we will use oe_content_document_field module, which will be available when project is merged.

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (ademarco) [inline modules/oe_content_tender/oe_content_tender.info.yml:1]:
```suggestion
name: OpenEuropa Call for Tender Content
```

File: modules/oe_content_tender/README.md
Comment (ademarco) [inline modules/oe_content_tender/README.md:1]:
```suggestion
# OpenEuropa Call for Tender Content
```

**Review by ademarco — CHANGES_REQUESTED** [review]

**Review by ademarco — APPROVED** [review]

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (yenyasinn) [inline modules/oe_content_tender/oe_content_tender.info.yml:14]: Todo has to be: `Set dependency oe_content:oe_content_documents_field instead of oe_content:oe_content_publication once project epic is merged!`

**Review by nagyad — APPROVED** [review]

File: modules/oe_content_tender/oe_content_tender.info.yml
Comment (ademarco) [reply modules/oe_content_tender/oe_content_tender.info.yml:14]: Yep we will fix this in the next one, so we merge this one and unblock the track.

**Pull request merged**
